src/Symfony/Component/Security/Guard/Authenticator/GuardBridgeAuthenticator.php

```diff
@@ -62,14 +63,11 @@ public function authenticate(Request $request): PassportInterface
         }
 
         // get the user from the GuardAuthenticator
-        $user = $this->guard->getUser($credentials, $this->userProvider);
-
-        if (null === $user) {
-            throw new UsernameNotFoundException(sprintf('Null returned from "%s::getUser()".', get_debug_type($this->guard)));
-        }
-
-        if (!$user instanceof UserInterface) {
-            throw new \UnexpectedValueException(sprintf('The "%s::getUser()" method must return a UserInterface. You returned "%s".', get_debug_type($this->guard), get_debug_type($user)));
+        if (class_exists(UserBadge::class)) {
+            $user = new UserBadge(null, function () use ($credentials) { return $this->getUser($credentials); });
```

Comment:
If this is the reason why we've made the first arg to `UserBadge` null, then we *could* work around that by using `'guard_credentials'.md5($credentials)` or something like that.